### ISSUE TITLE
[Feature]: Add support for global function deferral

### ISSUE THREAD

**Issue by tsv1** [vedro-universe/vedro#99]
**Summary:**

Currently, the `DeferrerPlugin` defers all functions to the end of each scenario, which is useful in many cases. However, certain use cases require deferring functions until the end of the entire test session, such as cleaning up shared resources that are used across multiple scenarios. This feature request proposes adding support for global deferral, where functions are executed after the entire test run, along with the ability to specify which queue (scenario-level or global-level) a deferred function should be added to.

**Proposed Solution:**

1. **Global Deferral Function:**
   Introduce a new function called `defer_global` (or `global_defer`, whichever is more intuitive) that adds a function to a global defer queue. This function would allow users to defer cleanup tasks or actions that should only occur once after all scenarios have completed.

   The `defer_global` function will behave similarly to `defer`, but instead of executing the deferred functions at the end of a scenario, it will defer execution until the end of the entire test session.

   ```python
   def defer_global(fn: Callable[..., Any], *args: Any, **kwargs: Any) -> None:
       """
       Defer the execution of a function until the end of the entire test session.

       The function will be added to a global queue and executed after all scenarios have completed.

       :param fn: The function to be deferred.
       :param args: Positional arguments to be passed to the function.
       :param kwargs: Keyword arguments to be passed to the function.
       """
       _global_queue.append((fn, args, kwargs))
   ```

2. **Global Queue Execution:**
   Modify the `DeferrerPlugin` to execute the global queue during the cleanup phase at the end of the test run (after all scenarios have been completed). This ensures that functions added to the global queue are executed in the correct order (LIFO) and only once at the end of the session.

3. **Use Cases:**
   The ability to globally defer functions would be useful in the following scenarios:
   
   - **Close shared browser instances:** If a browser instance (e.g., `Selenium`, `Playwright`) is shared across multiple tests for performance reasons, it can be closed at the end of the test session to speed up testing.
     ```python
     defer_global(browser.quit)
     ```
   
   - **Close HTTP sessions (async):** For asynchronous HTTP sessions (e.g., `aiohttp`, `httpx`), the session can be closed once at the end of the test run.
     ```python
     defer_global(http_session.close)
     ```
   
   - **Call singleton destructors:** If singleton resources or objects (such as database connections or external services) are used, their cleanup (like destructors) can be deferred to the end of the session to avoid reinitialization costs.
     ```python
     defer_global(singleton_instance.destroy)
     ```

**Example Usage:**

1. **Scenario-Level Deferral (default):**
   ```python
   defer(cleanup_temp_files)
   ```

   This will defer the `cleanup_temp_files` function to execute at the end of the current scenario.

2. **Global-Level Deferral:**
   ```python
   defer_global(close_browser)
   defer_global(http_session.close)
   ```

   These functions will be deferred to execute at the end of the entire test session, after all scenarios have completed.
